src/lib/mina.ts

```diff
@@ -786,86 +243,123 @@ function Network(
         `getNetworkState: Could not fetch network state from graphql endpoint ${minaGraphqlEndpoint} outside of a transaction.`
       );
     },
-    async sendTransaction(txn: Transaction) {
+    async sendTransaction(txn: Transaction): Promise<PendingTransaction> {
       txn.sign();
 
       verifyTransactionLimits(txn.transaction);
 
       let [response, error] = await Fetch.sendZkapp(txn.toJSON());
-      let errors: any[] | undefined;
+      let errors: string[] = [];
       if (response === undefined && error !== undefined) {
-        console.log('Error: Failed to send transaction', error);
-        errors = [error];
+        errors = [JSON.stringify(error)];
       } else if (response && response.errors && response.errors.length > 0) {
-        console.log(
-          'Error: Transaction returned with errors',
-          JSON.stringify(response.errors, null, 2)
-        );
-        errors = response.errors;
+        response?.errors.forEach((e: any) => errors.push(JSON.stringify(e)));
       }
 
-      let isSuccess = errors === undefined;
-      let maxAttempts: number;
-      let attempts = 0;
-      let interval: number;
-
-      return {
+      const isSuccess = errors.length === 0;
+      const hash = Test.transactionHash.hashZkAppCommand(txn.toJSON());
+      const pendingTransaction: Omit<
+        PendingTransaction,
+        'wait' | 'waitOrThrowIfError'
+      > = {
         isSuccess,
         data: response?.data,
         errors,
-        async wait(options?: { maxAttempts?: number; interval?: number }) {
-          if (!isSuccess) {
-            console.warn(
-              'Transaction.wait(): returning immediately because the transaction was not successful.'
+        transaction: txn.transaction,
+        toJSON: txn.toJSON,
+        toPretty: txn.toPretty,
+        hash() {
+          return hash;
+        },
```

Comment:
since hash is just a value now, maybe we could make this a property instead of function?

Reply:
done!